### PR TITLE
Reproduction

### PR DESCRIPTION
**Pull request by JesseBrouw**
Added reproducibility script (already merged with optimisation branch)

### CODE REVIEW

File: inpars/inpars.py [hunk 1]
```diff
@@ -1,3 +1,296 @@
+# import os
```
Comment (MJHamar) [inline inpars/inpars.py]: can we delete this commented block?

File: inpars/inpars.py [hunk 2]
```diff
@@ -9,10 +302,6 @@
 import urllib.error
 from .prompt import Prompt
 
-import json
```
Comment (MJHamar) [inline inpars/inpars.py:14]: do I understand correctly that this file was basically restored to the original version and inpars_plus is the optimized one?

File: src/reproduce.py
Comment (MJHamar) [inline src/reproduce.py:1]: looks neat!

**Review by MJHamar — COMMENTED** [review]
maybe we could get rid of that massive commented code block in inpars.py?

File: inpars/inpars.py (hunk 1 above)
Comment (JesseBrouw) [reply inpars/inpars.py]: That commented block is the updated inpars file generated by Danilo (with caching). I now use the regular InPars file for reproduction. We could remove It, but maybe save that one in a different file then

File: inpars/inpars.py (hunk 2 above)
Comment (JesseBrouw) [reply inpars/inpars.py:14]: Yes.

Comment (MJHamar) [reply inpars/inpars.py:14]: then the commented code in inpars.py is redundant no?

Comment (MJHamar) [reply inpars/inpars.py:14]: but I don't wanna sound to picky, it's up to you

Comment (JesseBrouw) [reply inpars/inpars.py:14]: I'm not sure whether it is exactly the same, I haven't compared them extensively. inpars_plus was created by Danilo just a few days ago, while the initial on (commented here) was already created last week

**Comment by MJHamar** [comment]
should we merge this?

**Comment by MJHamar** [comment]
should we get this on the main branch soon?

**Comment by JesseBrouw** [comment]
I created a separate file "inpars_w_caching.py" where the caching is implemented. There were quite some changes in inpars.py and I haven't used it for my reproduction experiments. I fixed compatibility with the repro branch, but just to be sure I also maintained the original inpars implementation.

**Pull request merged**